### ISSUE TITLE
Lines with gradients not showing

### ISSUE THREAD

**Issue by arjun010** [exupero/saveSvgAsPng#56]
Hi, I'm trying to use this library to download my graph visualization as a PNG. In my case, I have two types of coloring for the edges -- one where it is interpolated between the colors of the nodes it connects and the other where it is gray.

For the interpolation effect, I am creating several svg linear gradients and applying the appropriate one for each edge using the url() for the stroke property. When I try to download a network with interpolated edges, it doesn't download the edges, I can only see the nodes in the image.

In the case of non interpolated edges, the stroke is gray and that gets downloaded correctly. Could you please give me some tips on how to fix this?

**Comment by exupero** [comment]
Thanks for reporting this. Can you supply some example code that demonstrates this problem? I'd like to add a test case for it. Personally, I haven't tried gradients, but I would expect it to work so I don't know where it's breaking down.

**Comment by arjun010** [comment]
Hey, thanks for getting back. I've made a quick fiddle with just one line . It can be found here:  https://jsfiddle.net/arjun010/c5tyykjr/

I wasn't sure how to add the saveSvgAsPng file on jsFiddle ( I can attach a zip of the test project folder if you want me to do that)

Just to give you a little more info about something I tried, I thought the download was getting messed up because I had my <defs> tag outside the visualization svg. So I changed it and put it inside the svg. When I run that, I get the error:

```
Uncaught URIError: URI malformed
```

in the reEncode(data) function in your code. Not sure what I'm missing. I would really appreciate any help with respect to this.
